numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -262,22 +263,38 @@ def link_to_library_functions(
         self.reload_init = []
 
     def maybe_link_nrt(self, link, tgt_ctx, asm):
+        """
+        Add the NRT source code to the link if the neccesary conditions are met:
+        If NRT is enabled for the CUDATargetContext, return True if we either
+        detect .extern .func decls of NRT functions in any file in the link or
+        any of the passed LinkableCode objects have NRT enabled. In the special
+        case of PTXSource or CUSource objects, the source code will be inspected
+        even if the NRT flag is not set.
+        """
         if not tgt_ctx.enable_nrt:
             return
 
-        all_nrt = "|".join(self.NRT_functions)
-        pattern = (
-            r"\.extern\s+\.func\s+(?:\s*\(.+\)\s*)?("
-            + all_nrt
-            + r")\s*\([^)]*\)\s*;"
-        )
+        def nrt_in_asm(asm):
```

Comment:
A minor nit: the declaration of `nrt_in_asm` seems superfluous now that it's called exactly once, and the original code would seem to have been sufficient with on the addition of the checks of linkable code below.